geoportal/c2cgeoportal_geoportal/lib/oidc.py

```diff
@@ -96,6 +97,115 @@ class OidcRememberObject(TypedDict):
     roles: list[str]
 
 
+def get_remember_from_user_info(
+    request: pyramid.request.Request, user_info: dict[str, Any], remember_object: OidcRememberObject
+) -> None:
+    """
+    Fill the remember object from the user info.
+
+    The remember object will be stored in a cookie to remember the user.
+
+    :param user_info: The user info from the ID token or from the user info view according to the `query_user_info` configuration.
+    :param remember_object: The object to fill, by default with the `username`, `email`, `settings_role` and `roles`,
+        the corresponding field from `user_info` can be configured in `user_info_fields`.
+    :param settings: The OpenID Connect configuration.
+    """
+    settings_fields = (
+        request.registry.settings.get("authentication", {})
+        .get("openid_connect", {})
+        .get("user_info_fields", {})
+    )
+
+    for field_, default_field in (
+        ("username", "sub"),
+        ("display_name", "name"),
+        ("email", "email"),
+        ("settings_role", None),
+        ("roles", None),
+    ):
+        user_info_field = settings_fields.get(field_, default_field)
+        if user_info_field is not None:
+            if user_info_field not in user_info:
+                _LOG.error(
+                    "Field '%s' not found in user info, available: %s.",
+                    user_info_field,
+                    ", ".join(user_info.keys()),
+                )
+                raise HTTPInternalServerError(f"Field '{user_info_field}' not found in user info.")
+            remember_object[field_] = user_info[user_info_field]  # type: ignore[literal-required]
+
+
+def get_user_from_remember(
+    request: pyramid.request.Request, remember_object: OidcRememberObject, create_user: bool = False
+) -> Union["static.User", DynamicUser] | None:
+    """
+    Create a user from the remember object filled from `get_remember_from_user_info`.
+
+    :param remember_object: The object to fill, by default with the `username`, `email`, `settings_role` and `roles`.
+    :param settings: The OpenID Connect configuration.
+    :param create_user: If the user should be created if it does not exist.
+    """
+    from c2cgeoportal_commons import models  # pylint: disable=import-outside-toplevel
```

Comment:
can you add a comment here to explain why the models are imported here and not at the top level?